hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -1809,6 +1809,37 @@ def accept(*args, **kwargs):
     return accept
 
 
+@defines_strategy
+@composite
+def subsequence_of(
+    draw,  # type: ignore
+    elements,  # type: SearchStrategy[Ex]
+    min_size=None,  # type: int
+    average_size=None,  # type: int
+    max_size=None,  # type: int
+):
+    # type: (...) -> List[Ex]
+    """Returns a strategy which generates sub-sequences of the elements
+    sequence.
+    """
+    check_valid_sizes(min_size, average_size, max_size)
+
+    # TODO: fix hax to always get numerical constraints.
+    min_size = 0 if min_size is None else min_size
+    max_size = len(elements) if max_size is None else max_size
+
+    def element_mask():
```

Comment:
As currently written, this inner function doesn't add any value.  Two options:

1. Just delete it, and run the code directly.
2. Move the `@composite` decorator to an inner function, and `return inner_strategy()`.
  This sacrifices code clarity for performance, so IMO it's not worth it here (validating sizes is cheap).